Source/WebCore/page/LocalDOMWindow.cpp

```diff
@@ -1692,7 +1692,11 @@ Ref<CSSStyleDeclaration> LocalDOMWindow::getComputedStyle(Element& element, cons
     // FIXME: This does not work for pseudo-elements that take arguments (webkit.org/b/264103).
     if (pseudoElt.startsWith(":"_s))
         pseudoId = CSSSelector::parsePseudoElement(pseudoElt, CSSSelectorParserContext { element.document() });
-    return CSSComputedStyleDeclaration::create(element, pseudoId);
+    if (!pseudoId)
+        return CSSComputedStyleDeclaration::create(element, IsEmptyStyle::Yes);
+    if (*pseudoId == PseudoId::None)
+        return CSSComputedStyleDeclaration::create(element, std::nullopt);
+    return CSSComputedStyleDeclaration::create(element, Style::PseudoElementIdentifier { *pseudoId });
```

Comment:
Factoring of this function is now bit odd and hard to follow for such a small function. How about separating the startsWith(":") path completely?